indico/modules/events/editing/notifications.py

```diff
@@ -0,0 +1,82 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2020 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from indico.core.notifications import make_email, send_email
+from indico.modules.events.editing.schemas import EditingConfirmationAction
+from indico.web.flask.templating import get_template_module
+
+
+def notify_submitter_comment(revision):
+    """
+    Notify the editor about comments made by a submitter.
+    """
+    submitter = revision.submitter
+    editor = revision.editable.editor
+    if not editor:
+        return
+    tpl = get_template_module('events/editing/emails/submitter_comment_notification.txt',
+                              submitter_name=submitter.first_name, timeline_url=revision.editable.external_timeline_url,
+                              recipient_name=editor.first_name)
+    send_email(make_email(editor.email, template=tpl))
+
+
+def notify_editor_comment(revision, user):
+    """
+    Notify the submitter about comments made by an editor.
+    """
+    submitter = revision.submitter
+    tpl = get_template_module('events/editing/emails/editor_comment_notification.txt',
+                              editor_name=user.first_name,
+                              timeline_url=revision.editable.external_timeline_url,
+                              recipient_name=submitter.first_name)
+    send_email(make_email(submitter.email, template=tpl))
+
+
+def notify_editor_judgment(revision, user):
+    """
+    Notify the submitter about a judgement made by an editor.
+    """
+    submitter = revision.submitter
+    tpl = get_template_module('events/editing/emails/editor_judgement_notification.txt',
+                              editor_name=user.first_name,
+                              timeline_url=revision.editable.external_timeline_url,
+                              recipient_name=submitter.first_name)
+    send_email(make_email(submitter.email, template=tpl))
+
+
+def notify_submitter_upload(revision):
+    """
+    Notify the editor about the submitter uploading a new revision.
+    """
+    submitter = revision.submitter
+    editor = revision.editable.editor
+    tpl = get_template_module('events/editing/emails/submitter_upload_notification.txt',
+                              submitter_name=submitter.first_name,
+                              timeline_url=revision.editable.external_timeline_url,
+                              recipient_name=editor.first_name)
+    send_email(make_email(editor.email, template=tpl))
+
+
+def notify_submitter_confirmation(revision, action):
+    """
+    Notify the editor(s) about submitter accepting/rejecting revision changes.
+    """
+    submitter = revision.submitter
```

Comment:
This may not actually be the user performing the action. Pass the `user` (coming from `session.user`) through to know who is actually doing the action.